Shaily_StataHw1_10:2.do

```diff
@@ -0,0 +1,234 @@
+
+***PART C***
+*make a new variable to see if payouts are consistent within each village
+sort village payout 
+gen village_consistent = 1 
+replace village_consistent = 0 if village[_n]==village[_n+1] & payout[_n]==payout[_n+1]
+replace village_consistent =0 if  village[_n]==village[_n-1] & payout[_n]==payout[_n-1]
+*now finding which households are in different pixels and have different payout statuses
+gen village_pixel_payout = .
+replace village_pixel_payout = 1 if pixel_village == 0 
+replace village_pixel_payout = 2 if pixel_village == 1 & village_consistent == 0
+replace village_pixel_payout = 3 if pixel_village == 1 & village_consistent == 1
+count if village_pixel_payout == 2
+list hhid if village_pixel_payout == 2
+count if village_pixel_payout == 3
+list hhid if village_pixel_payout == 3
+
+
+/*******************************************************************************
+Q2: PSLE TANZANIA DATA CLEANING
+*******************************************************************************/
+use psle_student_raw.dta, clear
+
+split s, parse(">SUBJECTS")
+*right now all the data is in the same row, we need to convert so each row is a unique student
+split s2, parse("</TD></TR>") gen(var) //
+
+gen serial = _n
+
+save psle_full_edited, replace
+
+*run loop over each row (school)
+forvalue i=1/138{
+	use psle_full_edited.dta, clear
+	keep if serial == `i'
+	display as error `i'
+	reshape long var, i(serial) j(j)
+	split var, parse("</FONT></TD>")
+	*keep only the relevant variables
+	keep var1 var2 var3 var4 var5
+	*dropping first and last rows as they are empty
+	drop if var2=="" & var3==""
+	*candidate ID variable
+	gen cand_id = substr(var1,-14,.)
+	*gender
+	gen gender = substr(var3,-1,.)
+	*Prem Number
+	gen prem_number =  substr(var2,strpos(var2, "CENTER") +8, .)
+	*Name
+	gen name =  substr(var4,strpos(var4, "<P>") +3 , .)
+	*grades
+	replace var5 = substr(var5,ustrpos(var5, "LEFT") +6 , .)
+	replace var5 = substr(var5,1 , strlen(var5) - 7)
+	*School Code
+	gen schoolcode = `i'
+	
+	*all the subject info is in one column, create separate columns
+	split var5, parse(,) 
+	*rename variables
+	rename var51 kiswahili
+	rename var52 english
+	rename var53 maarifa
+	rename var54 hisabati
+	rename var55 science
+	rename var56 uraia
+	rename var57 average
+	
+	*drop columns that are no longer needed.
+	drop var1 var2 var3 var4 var5 
+	*extract just the grade
+	local varlist kiswahili english maarifa hisabati science uraia average
+	foreach subject in `varlist'{
+		replace `subject' = usubstr(`subject',-1,1)
+		}
+	
+	*unsure about this part
+	tempfile school`i'
+	save `school`i''
+	append using `school`i''
+	save psle_final, replace
+}
+
```

Comment:
Good work on Q2. I wonder if you got a chance to create the school id. Just based on eyeballing the data, seems like it is the portion of the candidate id before the hyphen. The rest looks good!